src/components/inputs/money-input/money-input.js

```diff
@@ -185,15 +182,15 @@ export const parseRawAmountToNumber = rawAmount => {
 //  - no currency was selected
 //
 // This function expects the "amount" to be a trimmed value.
-export const createMoneyValue = (currencyCode, rawAmount) => {
+export const createMoneyValue = (currencyCode, rawAmount, locale) => {
   if (!currencyCode) return null;
 
   const currency = currencies[currencyCode];
   if (!currency) return null;
 
   if (rawAmount.length === 0 || !isNumberish(rawAmount)) return null;
 
-  const amountAsNumber = parseRawAmountToNumber(rawAmount);
+  const amountAsNumber = parseRawAmountToNumber(rawAmount, locale);
```

Comment:
This would be a breaking change as it stands? I may also not be following along adequately. But Can't the passed `locale` be considered a hint. So it works without but if passed, needed in certain circumstances, it ensures that parsing covers the last x% which otherwise fail?

Reply:
Yes this would be a breaking change.
That's a good point. We can use the previous logic if locale is not passed, but this can cause incorrect value for fractionless currencies.
given that we can:
Add invariant that locale is required if the currency has no fractions,
and if locale isn't passed and currency has fraction digits use previous logic.
WDUT?

Reply:
To we know the cases where it won't work beforehand or can it happen any time with MoneyInput and just depends on what the user has defined. Depending on that this might have to be breaking and we should always pass the locale.

Reply:
We know that the functions `MoneyInput.convertToMoneyValue` and `MoneyInput.isHighPrecision` will always fail if the currency has no fraction digits.

Reply:
Added an `invariant` that will throw if the currency has no fraction digits, and no local is provided

Reply:
Cool. Makes sense.

Reply:
Will this now be considered breaking change?

Reply:
Not to what I can tell.